libtuf/src/main/scala/com/advancedtelematic/libtuf/crypt/TufCrypto.scala

```diff
@@ -278,25 +310,31 @@ protected [crypt] class RsaCrypto extends TufCrypto[RsaKeyType.type] {
 
     Try {
       val pemKeyPair = parser.readObject().asInstanceOf[PEMKeyPair]
-      RSATufPrivateKey(converter.getKeyPair(pemKeyPair).getPrivate)
+      convertPrivate(converter.getKeyPair(pemKeyPair).getPrivate)
     }
   }
 
   override def encode(keyVal: RSATufKey): Json = Json.fromString(keyVal.keyval.toPem)
 
   override def encode(keyVal: RSATufPrivateKey): Json = Json.fromString(keyVal.keyval.toPem)
 
-  override def signer: security.Signature =
-    java.security.Signature.getInstance("SHA256withRSAandMGF1", "BC") // RSASSA-PSS
+  override def signer: security.Signature = java.security.Signature.getInstance("SHA256withRSAandMGF1", "BC") // RSASSA-PSS
 
```

Comment:
Maybe it's worth defining a SignatureAlgorithm for those, as well?

Reply:
Both curves work with this signature algorithm, as far as I could tell, whad do you mean?

Reply:
I mean adding it here as a new case class, looks the same to me: https://github.com/advancedtelematic/ota-tuf/pull/148/files#diff-677306bca67401daeb50547b1e350a12R233

Reply:
`SignatureAlgorithm` is for EC crypto (we support 2 different EC curves), this one is for an RSA key.